### DISCUSSION TITLE
Show connection libre2 in graph

### DISCUSSION THREAD

**Discussion by miguelavh** [NightscoutFoundation/xDrip#2554] [General]
Hi, is there any option to see connection state of libre2 sensor in graph? I have a lot of disconnection and I want to see when it happens... Thank you.

**Comment by PhilTU** [comment]
Variant https://github.com/jakobheyman/xdripJH displays every value (1 minute) so you would see gap ??

**Reply by Navid200** [reply]
That's not a variant.  
Variant is an official release from this repository.

People can create a fork of any repository.  
But, please don't use this repository and these discussions for suggesting someone else's fork.

Please work with us.  If something needs to change, please suggest.
If the change you want is not approved, there must be a reason.  You can state your opinion and explain why it should be merged.  But, if it is not, this repository remains the place to talk about our goals and our plans and our limitations.

**Reply by PhilTU** [reply]
OED begs to differ, but I take your point that "xDrip+ variant" has a particular meaning.

I shall think separately about how to address the bad bluetooth connectivity and limited diagnostics ....

**Reply by Navid200** [reply]
It doesn't matter what you choose to use.  We cannot force you.  And we don't try.

But, this repository and its resources are meant to support this repository.  
If we have made a wrong decision and have not merged a PR, please work with us to resolve any outstanding requests.  
If you choose to use the PR even though it has not been merged, you can do whatever you like.
But, please don't post a link in these discussions to a PR or fork presenting it like it is a solution for the community.

**Reply by PhilTU** [reply]
I posted a link that addresses the OP's issue. If we have to be constrained in what we discuss then please try the following :- 

Can you please provide a continuous watchdog facility to indicate to the user when the signal from a Libre 2 sensor has been lost. This should indicate if it is more than 70 seconds since the last glucose value was received and be capable of being permanently visible on the main screen and any widget, lock screen or watch display panel.

Can you also add a value for RSSI in dBm or similar to the Bluetooth status panel that will indicate to the user the Bluetooth signal strength last received or measured by the operating system, if this data is available. This will help diagnose orientation or location issues and also identify failure of the sensor Bluetooth signal at the radio level (which may be different to not receiving a useful glucose value).

**Reply by tzachi-dar** [reply]
[PhilTU](https://github.com/PhilTU) why wouldn't xdrip in your version close the gap at a later time? For example if there was no data for 5 minutes, than later when more data comes, xDrip will solve the gaps using later data.
[miguelavh](https://github.com/miguelavh) do you want to see this in real time or after some time has passed?

**Reply by PhilTU** [reply]
@tzachi-dar I am used to graphs with gaps in, and hadn't noticed any backfill happen over Bluetooth from a Libre 2.

I only suggested the "package that shall not be mentioned" as it displays 1 minute values, making it easy to see when the signal stops. This is useful if for example you go into a room and it stops, or connect an accessory or open an app, then you can correlate the loss of glucose values with an action or event.

**Reply by tzachi-dar** [reply]
I guess that your version is based on very old xdrip versions. Newer versions of xDrip have the close gap capabilities for ~1 hour.
You can also look at the "show libre trend" graph and see that you will get gaps closed up to 12 minutes ago.
This should also happen if you have the show libre points on the display set on.

**Reply by PhilTU** [reply]
@tzachi-dar it isn't "my" version, I gave it a try as it talked of 1 minute values when there was a discussion about how everything in xDrip is based on 5 minute values. The classic status page of the unmentionable says "Downgradable to 1603091400" same as my current standard version e13a76e-2022.08.03 (I don't know the significance of that).

Which option in current xDrip will give a 1 minute value on the graph - I note that "Show Libre Trend" and "Show backfilled data" default to off.

Perhaps there is already a solution with the right combination of options - can someone spell these out please so that one dot goes on the xDriup+ graph for every 1 minute value received fromt he Libre 2 sensor via OOP2 ??
